### ISSUE TITLE
Select does not position dropdown correctly

### ISSUE THREAD

**Issue by Michael-Tajmajer-Bentley** [iTwin/iTwinUI-react#17]
I am trying to use the Select control inside of a modeless dialog. I have found that when I use the dropdown, the list is not positioned correctly.

I have attached a gif of this behavior.

I am using the Dialog from the @bentley/ui-core, and opening it via the ModelessDialogManager.openDialog().

This dialog is quite simple:

    <Dialog
      title={translate("networkTrace:AdvancedDialog.Title")}
      opened
      modal={false}
      modelessId={dialogId}
      onModelessPointerDown={(event) =>
        ModelessDialogManager.handlePointerDownEvent(event, dialogId, forceUpdate)}
      style={{
        zIndex: ModelessDialogManager.getDialogZIndex(dialogId),
      }}
      movable
      width={450}
      height={220}
      onClose={closeDialog}
      onEscape={closeDialog}
      buttonCluster={[
        { type: DialogButtonType.OK, onClick: handleOK, label: translate("networkTrace:AdvancedDialog.RunTrace") },
        { type: DialogButtonType.Cancel, onClick: closeDialog, label: translate("networkTrace:AdvancedDialog.Close") },
      ]}
    >
      <Select<number>
        options={options}
        value={value}
        onChange={setValue}
        placeholder={translate("networkTrace:AdvancedDialog.SelectPlaceholder")}
      />
    </Dialog>

Any thoughts on how to correct this?

![select-positioning](https://user-images.githubusercontent.com/58269392/115607557-bb924f00-a2aa-11eb-9569-9549edbe6bf4.gif)

**Comment by mayank99** [comment]
Are you able to reproduce a minimal example in a code sandbox? (fork [this](https://codesandbox.io/s/itwinui-react-example-son74) and add `@bentley/ui-core` as dependency).

Also can you provide some environment info? Package versions, OS version, browser version, etc (all of which are part of the issue template).

**Comment by Michael-Tajmajer-Bentley** [comment]
Where is the issue template located? I looked around for it, but can't seem to locate it :-( would you provide a link?

Package versions:
"@itwin/itwinui-react": "1.1.0"
"@bentley/ui-core": "2.14.1" 
"@bentley/ui-framework": "2.14.1" (all the @bentley packages are 2.14.1)

Browser/os: 
Chrome 90.0.4430.85
Windows 10 - 20H2

I've forked the example and have added ui-core and ui-framework to it. After that I get an error in the console:
![image](https://user-images.githubusercontent.com/58269392/115619090-17fc6b00-a2b9-11eb-8d2c-c4f3b71c67b1.png)

https://codesandbox.io/s/itwinui-react-example-forked-0i9h3?file=/src/App.tsx:257-278

I added the ui-framework to allow me to use the ModelessDialogManager to open the dialog - which is where I am seeing the issue.

Any thoughts on how to proceed? I guess I could create an itwin-viewer-react example outside of the sandbox if that would be faster - thoughts?

**Comment by mayank99** [comment]
Thanks for the info. For future reference, the issue template shows up when you create a new issue and choose "bug report" from https://github.com/iTwin/iTwinUI-react/issues/new/choose

That error looks very cryptic. My best guess is that it's missing some dependency or it needs additional TS configuration (e.g. CJS vs ESM)

Yeah, you could create an itwin-viewer sandbox if it would be easier. Ultimately, what we really need is to be able to reproduce the issue on our side to proceed with debugging.

**Comment by Michael-Tajmajer-Bentley** [comment]
Yup, as I suspected I pressed the wrong button :-) - thank you for pointing it out, I'll be sure to use it if I need to report anything in the future.

In that case, I'll see if I can get an itwin-viewer-react based example put together.

**Comment by Michael-Tajmajer-Bentley** [comment]
I created a simple-ish sample that demonstrates the issue: https://github.com/Michael-Tajmajer-Bentley/iTwinUI-react-select-issue

It needs a model & client id to run (see the .env file).

**Comment by mayank99** [comment]
After several hours of debugging, here's what we've found:

The ui-core `Dialog` has some css rules that conflict with the `transform` calculations needed to position our Select menu.

In particular, the `.core-dialog-hidden` class used here is causing problems because it forces the parent of the Select menu to have 0x0 dimensions. Ideally, the "no overlay" styling would be separate from the dialog functionality in a way that it does not need 0x0 elements. Is that something that's possible? @DanEastBentley @bsteinbk
https://github.com/imodeljs/imodeljs/blob/289a1dc91a0556d559f3c4fdc1093ef1bc73e3f2/ui/core/src/ui-core/dialog/Dialog.tsx#L273

As a workaround, #22 is adding `popoverProps` to `Select`. This would allow you to append the menu to a different DOM element outside the `Dialog`, but you will need to set a higher z-index. Does that work for you? @Michael-Tajmajer-Bentley 
<details>
<summary>See code snippet</summary>

```jsx
<Select
  {...selectProps} // value, options, etc
  popoverProps={{
    appendTo: () => document.body,
    zIndex: 13002,
  }}
/>
```
</details>

**Comment by Michael-Tajmajer-Bentley** [comment]
This change will be fine for me - thanks @mayank99, your work is appreciated! 

When do you think a new build would be ready? I'm not blocked by this, just curious.

**Comment by mayank99** [comment]
Next release should happen in the next few days. Maybe later this week or early next week. I will update this thread when it happens.

**Comment by veekeys** [comment]
Hello @Michael-Tajmajer-Bentley . Thank you for the finding! As @mayank99 mentioned, we have found a possibility how you could have it fixed. Just would be great, not to close the issue, before you actually get the fix :D It kinda conflicts our workflow.

**Comment by DanEastBentley** [comment]
I have submitted a PR to fix the modeless dialog issue in iModel.js: https://github.com/imodeljs/imodeljs/pull/1267

**Comment by mayank99** [comment]
@Michael-Tajmajer-Bentley We just released [v1.2.0](https://github.com/iTwin/iTwinUI-react/releases/tag/v1.2.0) where you can use the workaround I mentioned above.

**Comment by Michael-Tajmajer-Bentley** [comment]
Dan,

Tried out the new build and things are fine now – thank you for your efforts in getting this resolved!

Regards,

Mike Tajmajer
